packages/flutter/lib/src/material/data_table.dart

```diff
@@ -0,0 +1,757 @@
+  _SortArrow({
+    Key key,
+    this.visible,
+    this.down,
+    this.duration
+  }) : super(key: key);
+
+  final bool visible;
+
+  final bool down;
+
+  final Duration duration;
+
+  @override
+  _SortArrowState createState() => new _SortArrowState();
+}
+
+class _SortArrowState extends State<_SortArrow> {
+
+  AnimationController _opacityController;
+  Animation<double> _opacityAnimation;
+
+  AnimationController _orientationController;
+  Animation<double> _orientationAnimation;
+  double _orientationOffset = 0.0;
+
+  bool _down;
+
+  @override
+  void initState() {
+    super.initState();
+    _opacityAnimation = new CurvedAnimation(
+      parent: _opacityController = new AnimationController(
+        duration: config.duration
+      ),
+      curve: Curves.ease
+    )
+    ..addListener(_rebuild);
+    _opacityController.value = config.visible ? 1.0 : 0.0;
+    _orientationAnimation = new Tween<double>(
+      begin: 0.0,
+      end: math.PI
+    ).animate(new CurvedAnimation(
+      parent: _orientationController = new AnimationController(
+        duration: config.duration
+      ),
+      curve: Curves.easeIn
+    ))
+    ..addListener(_rebuild)
+    ..addStatusListener(_resetOrientationAnimation);
+    if (config.visible)
+      _orientationOffset = config.down ? 0.0 : math.PI;
+  }
+
+  void _rebuild() {
+    setState(() {
+      // The animations changed, so we need to rebuild.
+    });
+  }
+
+  void _resetOrientationAnimation(AnimationStatus status) {
+    if (status == AnimationStatus.completed) {
+      assert(_orientationAnimation.value == math.PI);
+      _orientationOffset += math.PI;
+      _orientationController.value = 0.0; // TODO(ianh): This triggers a pointless rebuild.
+    }
+  }
+
+  @override
+  void didUpdateConfig(_SortArrow oldConfig) {
+    super.didUpdateConfig(oldConfig);
+    bool skipArrow = false;
+    bool newDown = config.down != null ? config.down : _down;
+    if (oldConfig.visible != config.visible) {
+      if (config.visible && (_opacityController.status == AnimationStatus.dismissed)) {
+        _orientationController.stop();
+        _orientationController.value = 0.0;
+        _orientationOffset = newDown ? 0.0 : math.PI;
+        skipArrow = true;
+      }
+      if (config.visible) {
+        _opacityController.forward();
+      } else {
+        _opacityController.reverse();   
+      }
+    }
+    if ((_down != newDown) && !skipArrow) {
+      if (_orientationController.status == AnimationStatus.dismissed) {
+        _orientationController.forward();
+      } else {
+        _orientationController.reverse();
+      }
+    }
+    _down = newDown;
+  }
+
+  @override
+  void dispose() {
+    _opacityController.dispose();
+    _orientationController.dispose();
+    super.dispose();
+  }
+
+  static const double _kArrowIconBaselineOffset = -1.5;
+  static const double _kArrowIconSize = 16.0;
+
+  @override
+  Widget build(BuildContext context) {
+    return new Opacity(
```

Comment:
I wish, but I have to control the opacity animation way more closely than FadeTransition allows.